useragent/useragent.go

```diff
@@ -123,7 +123,7 @@ func (a *Agent) getUserAgent(agent string, callTree string) (userAgent string) {
 	return
 }
 
-func checkForCIEnvironment() bool {
+func CheckForCIEnvironment() bool {
```

Comment:
I kinda think we should move this to another package? Not entirely sure WHERE but it's clear it doesn't belong to `useragent` anymore. WDYT?

Reply:
Sure, but I couldn't see a better "WHERE" either, so just left it. Open to ideas.

Reply:
Yeah I know, I'm like 🤯 

The best thing to do until we figure out where it goes, maybe move it to `internal/ci.go` or something like that, something at the root, so we can import it across packages but not make it a part of the public API?

Reply:
https://dave.cheney.net/2019/10/06/use-internal-packages-to-reduce-your-public-api-surface

Reply:
Moved some goodies to `internal`: 1bba8f7